itest/lnd_funding_test.go

```diff
@@ -780,15 +837,18 @@ func testBatchChanFunding(ht *lntest.HarnessTest) {
 
 	// Check if the change type from the batch_open_channel funding is P2TR.
 	rawTx := ht.Miner.GetRawTransaction(txHash)
-	require.Len(ht, rawTx.MsgTx().TxOut, 4)
+	require.Len(ht, rawTx.MsgTx().TxOut, 5)
 
 	// For calculating the change output index we use the formula for the
 	// sum of consecutive of integers (n(n+1)/2). All the channel point
 	// indexes are known, so we just calculate the difference to get the
 	// change output index.
-	changeIndex := uint32(6) - (chanPoint1.OutputIndex +
-		chanPoint2.OutputIndex + chanPoint3.OutputIndex)
-
+	// Example: Batch outputs = 4, sum_consecutive_ints(4) = 10
+	// Subtract all other output indices to get the change index:
+	// 10 - 0 - 1 - 2 - 3 = 4
+	changeIndex := uint32(10) - (chanPoint1.OutputIndex +
+		chanPoint2.OutputIndex + chanPoint3.OutputIndex +
+		chanPoint4.OutputIndex)
```

Comment:
Is this because the OutputIndex selection is nondeterministic? Otherwise this just seems much more complicated than necessary.

Reply:
I think we do BIP-0069 sorting of inputs and outputs so the channel points could be non-deterministic. And this algorithm (which I was confused about initially too) is a nice way to find out the remaining index after subtracting all of the others.

Reply:
It's a rather elegant solution to a problem I wasn't aware of was even a problem xD